src/shelter/shelter.service.ts

```diff
@@ -156,7 +179,21 @@ export class ShelterService {
     });
 
     const shelterStatusFilter = this.addShelterStatusFilter(payload);
-    const where = this.mountWhereFilter(payload);
+
+    const unnaccentShelterIds = await this.getUnaccentShelterIds(
+      payload.search,
+    );
+
+    if (payload.search && unnaccentShelterIds.length === 0) {
```

Comment:
Existem mais elementos de busca além do search. Sendo assim, poderia remover todo esse if e return?

Reply:
@AlbuquerqueRafael os filtros não são combinados? se a pessoa digitar: "batata" e nenhum resultado for encontrado, os demais filtros não devem importar

Reply:
@AlbuquerqueRafael esse if serve pra um caso específico, onde existe uma tentativa de busca, ou seja, o payload.search contém uma string, porém nada foi encontrado no banco na query. Sem esse if o comportamento era diferente do atual/esperado. Referenciei aqui: https://github.com/SOS-RS/backend/pull/30#issuecomment-2105141242

Reply:
update: dps de resolver os ultimos merge conflicts nao foi mais necessário manter esse if, por conta das alterações que subiram pra dev